tremor-connectors-azure/src/monitor/ingest/writer/sink.rs

```diff
@@ -0,0 +1,296 @@
+            config: None,
+        }),
+        CodecReq::Required,
+        vec![],
+        &ConnectorType::from("azure_monitor_ingest_writer"),
+        &tremor_common::alias::Connector::new(
+            "azure_monitor_ingest_writer", // TODO This doesn't adapt well to structured where codecs/processors are needed
+            "azure_monitor_ingest_writer",
+        ),
+    )
+    // ALLOW: We know at compile time that codecs and processors will resolve ok
+    .unwrap()
+}
+
+#[async_trait::async_trait]
+impl Sink for AmiSink {
+    async fn connect(&mut self, _ctx: &SinkContext, _attempt: &Attempt) -> anyhow::Result<bool> {
+        let azure_client = azure_core::new_http_client();
+        self.client = Some(azure_client);
+
+        Ok(true)
+    }
+
+    #[allow(clippy::too_many_lines)]
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        event: Event,
+        ctx: &SinkContext,
+        _serializer: &mut EventSerializer,
+        start: u64,
+    ) -> anyhow::Result<SinkReply> {
+        let guard = self.concurrency_cap.inc_for(&event);
+
+        if let Some(client) = self.client.clone() {
+            let task_ctx = ctx.clone();
+
+            let response_tx = self
+                .source_is_connected
+                .load(Ordering::Acquire)
+                .then(|| self.response_tx.clone());
+            let reply_tx = self.reply_tx.clone();
+
+            let contraflow_data = if event.transactional {
+                Some(ContraflowData::from(&event))
+            } else {
+                None
+            };
+
+            let mut origin_uri = self.origin_uri.clone();
+
+            // assign a unique request id to this event
+            let request_id = RequestId::new(self.request_counter);
+            self.request_counter = self.request_counter.wrapping_add(1).max(1);
+
+            let mut builder = None;
+            let mut correlation_meta = None;
+
+            for (value, meta) in event.value_meta_iter() {
+                match &mut builder {
+                    None => {
+                        // take the metadata from the first element of the batch
+                        correlation_meta = meta.get("correlation").map(Value::clone_static);
+                        let http_meta = ctx.extract_meta(meta);
+                        let mut once_per_request =
+                            RequestBuilder::new(request_id, http_meta, &self.config).await?;
+                        once_per_request
+                            .append(value, meta, start, &mut self.serializer)
+                            .await?;
+                        builder = Some(once_per_request);
+                    }
+                    Some(builder) => {
+                        builder
+                            .append(value, meta, start, &mut self.serializer)
+                            .await?;
+                    }
+                }
+            }
+
+            assert!(builder.is_some(), "No builder");
```

Comment:
This will panic the runtime, we want to use an error instead, we can completely remove this since builder.some is checked in line 179 and will error corretly

Reply:
That was debugging I forgot to cleanup. Whacked and well spotted...